src/hooks/useMainProcess.ts

```diff
@@ -33,9 +32,14 @@ export interface ExecutionInput {
   offrampAmount: Big;
 }
 
+interface UseMainProcessProps {
+  checkAndWaitForSignature: () => Promise<SiweSignatureData>;
+  forceRefreshAndWaitForSignature: () => Promise<SiweSignatureData>;
+}
+
 type ExtendedExecutionInput = ExecutionInput & { stellarEphemeralSecret: string };
 
-export const useMainProcess = () => {
+export const useMainProcess = ({ checkAndWaitForSignature, forceRefreshAndWaitForSignature }: UseMainProcessProps) => {
```

Comment:
Since `useMainProcess` is a hook itself, you can call other hooks in it. Thus, you don't need to pass these two functions as props but you can just call `useSignChallenge` inside the body of `useMainProcess` and access the respective functions.

Reply:
The thing is that we need `requiresSign`, `handleCancel`, etc for the modal component, which is on the same [level](https://github.com/pendulum-chain/vortex/pull/257/files#diff-777e0c2221f5feeef076c359496045ddd95d6d376a0fc1395b330f8a9b64a5bbR352) as `useMainProcess`. 

And those 2 functions as a [parameter](https://github.com/pendulum-chain/vortex/pull/257/files#diff-fedb8a30074ce22b1fb6aa02c40daa53038ea2a73c885a7bd7db6f5dfe5e8efaR152) in the `sep-10` which is inside `useMainProcess`. Alternatively we can set a context but seemed like more trouble, or return the modal states from `useMainProcess`.

I already faced the issue of using `useSignChallenge` on both levels and doesn't work of course.

Reply:
Then we might want to move parts of the `useSignChallenge` hook into a context so that it's shared and every call to `useSignChallenge` returns the shared values. A similar example would be the `useBridgeSettings` hook in the portal that [uses a shared context](https://github.com/pendulum-chain/portal/blob/44f8a2ff8743f31dd1665a394f7310d60152424a/src/hooks/spacewalk/useBridgeSettings.ts#L72) for storing the settings a user chose between issue and redeem requests.

Reply:
Moved to a [context](https://github.com/pendulum-chain/vortex/pull/257/commits/22f23390e28bbcb3393dc5eacec4fa8ad5f68486), I followed the `events` one which I found quite clean. This is what you had in mind?

Is this more or less what you had in mind? I don't think we can move only partially the hook into a context.